lib/core/src/Cardano/Wallet/Primitive/Slotting.hs

```diff
@@ -609,3 +619,32 @@ unsafeExtendSafeZone = f . neverFails r
         , handleResult = h
         }
     r = "unsafeExtendSafeZone should make PastHorizonExceptions impossible."
+
+{-------------------------------------------------------------------------------
+                       For testing
+-------------------------------------------------------------------------------}
+
+-- | A dummy 'TimeInterpreter' for testing. A horizon at @SlotNo 40@ allows
+-- testing 'PastHorizonException's.
+dummyForkInterpreter :: TimeInterpreter (Either PastHorizonException)
+dummyForkInterpreter =
```

Comment:
Note: the difference between this `dummyForkInterpreter` and the `dummyTimeInterpreter` defined in `TransactionSpec` is that this has a limited horizon, but the `dummyTimeInterpreter` defined in `TransactionSpec` does not.